src/accelerate/accelerator.py

```diff
@@ -646,6 +660,26 @@ def get_state_dict(self, model):
 
         return state_dict
 
+    def register_for_checkpointing(self, *objects):
+        """
+        Makes note of `objects` and will save or load them in during `save_state` or `load_state`.
+
+        These should be utilized when the state is being loaded or saved in the same script. It is not designed to be
+        used in different scripts
+
+        Note: Every `object` must have a `load_state_dict` and `state_dict` function to be stored.
```

Comment:
This is not a proper docstring **but** it will be once #271 is merged, so giving you a free pass ;-)